### PR TITLE
Add example for download with progress tracking

### PR DESCRIPTION
**Pull request by Songtronix**

### CODE REVIEW

File: examples/download_progress/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+## Download Progress
+
+Downloading a file asynchronously with a `Subscription` while displaying the progress with a `ProgressBar`.
+
+<div align="center">
+  <a href="https://gfycat.com/watchfulwelcomeladybird">
+    <img src="https://thumbs.gfycat.com/InfamousIcyErmine-small.gif">
```
Comment (Songtronix) [inline examples/download_progress/README.md]: Forgot to update I guess.
```suggestion
    <img src="https://gfycat.com/watchfulwelcomeladybird">
```

Comment (Songtronix) [reply examples/download_progress/README.md]: Fixed.

**Comment by Songtronix** [comment]
If someone has an idea on how to integrate proper error handling into that I am surely interested!

**Comment by hecrj** [comment]
I am working on an improved version.

**Comment by hecrj** [comment]
I have updated the example to use `reqwest`, which implements `Response::chunk` and allows us to track progress more elegantly.

I also changed the example code a bit and renamed some things. Let me know what you think!

Download progress - Iced

**Comment by Songtronix** [comment]
I love it! :blue_heart:

**Pull request merged**